### DISCUSSION TITLE
Startup speed

### DISCUSSION THREAD

**Discussion by hoijui** [ClementTsang/bottom#1099] [Q&A] [answered]
Hey! :-)
just tried bottom for the first time now. I am used to `htop`. I am a rust fanboy and write in rust myself whenever I can.
I installed bottom through cargo, and noticed that.. it takes much longer to startup then htop, and was wondering why. or if that is normal. It is almost a second for me, which with htop it is maybe 0.2s or so (very subjective .. no measuring done here).
I also tried `--basic`, but that seemed not to make a difference.
Not to say that this is the end of the world. or that I could not get used to it, was just wondering.

**Comment by ClementTsang** [comment]
There is a small delay on startup to initialize some stuff, so that's *probably* normal. I think the UI should be drawn relatively quickly, but there is an intentional small delay before we start gathering and showing data. I can definitely take a look at seeing if this can be reduced this though, some of the code around that is super old.

If it's waiting for like 5+ seconds or something like that though, then no, that's not normal.

**Comment by hoijui** [comment]
wow.. thanks for the fast response! :-)
I am talking about the time it takes till something gets drawn at all (the screen is not black anymore). I think I would use `--basic` anyway.
.. hmm... now I feel it is only half a second till it draws something.

**Reply by ClementTsang** [reply]
Ah, yeah, it's probably due to some initialization code then (e.g. reading from configs, setting up event threads, etc.); I can see if I can reduce that, especially since I've been planning on doing some optimization passes before the next release.

**Reply by hoijui** [reply]
If I compare it to `lazygit`, it is very noticeably slower to draw something.
... this reminds me on the Java thing from many years back. everyone always said Java is slow, for many, many years, and when investigated, it turned out this comes from the time it took until the UI is drawn, for GUI applications. At first, a gray rectangle was shown, and some fractions of a second later, the UI elements would be drawn.
in short ... it might be possible that from a users perspective, it would be better to show the UI earlier, even if it meant sacrificing time till first gathering of data or something like that.

**Reply by hoijui** [reply]
thanks a lot! :-)

**Reply by ClementTsang** [reply, marked as answer]
Still need to test it a bit but https://github.com/ClementTsang/bottom/pull/1100 will hopefully make things feel a bit snappier.

**Reply by hoijui** [reply]
wo-hoo!! :D
indeed... no noticeable delay anymore until the UI gets drawn!
now one can see what you said, that it takes a short time till data starts arriving, but feels good now! :-)
thanks a lot @ClementTsang !

**Comment by Rhaletskyj** [comment]
thanks for the repository
